Build failure: LibreOffice

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

build on unstable small

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/311282458

### Relevant log output

```console
'libreoffice-25.2.6.2/external/tarballs/7191c669bf38899f73a2094ed00f7b800553364f90e2637010a69c0e268f25d0-liberation-fonts-ttf-2.1.5.tar.gz' -> '/nix/store/8a6c2v046a973f2kkcyxji3iqsziykvc-liberation-fonts-ttf-2.1.5.tar.gz'
...skipping...
      |                          ^~~~~~~~~
/nix/store/ffrg0560kj0066s4k9pznjand907nlnz-gcc-14.3.0/include/c++/14.3.0/optional:735:17: note: candidate: 'constexpr std::optional<_Tp>::optional() [with _Tp = GooString]'
  735 |       constexpr optional() noexcept { }
      |                 ^~~~~~~~
/nix/store/ffrg0560kj0066s4k9pznjand907nlnz-gcc-14.3.0/include/c++/14.3.0/optional:735:17: note:   candidate expects 0 arguments, 1 provided
/nix/store/ffrg0560kj0066s4k9pznjand907nlnz-gcc-14.3.0/include/c++/14.3.0/optional:703:11: note: candidate: 'constexpr std::optional<GooString>::optional(std::optional<GooString>&&)'
  703 |     class optional
      |           ^~~~~~~~
/nix/store/ffrg0560kj0066s4k9pznjand907nlnz-gcc-14.3.0/include/c++/14.3.0/optional:703:11: note:   no known conversion for argument 1 from 'GooString*' to 'std::optional<GooString>&&'
In file included from /nix/store/ffrg0560kj0066s4k9pznjand907nlnz-gcc-14.3.0/include/c++/14.3.0/memory:78,
                 from /nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:46,
                 from /nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/Object.h:45:
/nix/store/ffrg0560kj0066s4k9pznjand907nlnz-gcc-14.3.0/include/c++/14.3.0/bits/unique_ptr.h: In instantiation of 'std::__detail::__unique_ptr_t<_Tp> std::make_unique(_Args&& ...) [with _Tp = GooString; _Args = {GooString*&}; __detail::__unique_ptr_t<_Tp> = __detail::__unique_ptr_t<GooString>]':
/build/libreoffice-25.2.6.2/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx:185:45:   required from here
  185 |     PDFDoc aDoc( std::make_unique<GooString>(pFileName),
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/nix/store/ffrg0560kj0066s4k9pznjand907nlnz-gcc-14.3.0/include/c++/14.3.0/bits/unique_ptr.h:1077:30: error: no matching function for call to 'GooString::GooString(GooString*&)'
 1077 |     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:92:5: note: candidate: 'GooString::GooString(const std::string&, int, size_t)'
   92 |     GooString(const std::string &str, int idx, size_t lengthA) : std::string(str, idx, lengthA) { }
      |     ^~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:92:5: note:   candidate expects 3 arguments, 1 provided
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:91:5: note: candidate: 'GooString::GooString(const GooString*, int, size_t)'
   91 |     GooString(const GooString *str, int idx, size_t lengthA) : std::string(*str, idx, lengthA) { }
      |     ^~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:91:5: note:   candidate expects 3 arguments, 1 provided
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:88:5: note: candidate: 'GooString::GooString(const char*, size_t)'
   88 |     GooString(const char *sA, size_t lengthA) : std::string(sA ? sA : "", sA ? lengthA : 0) { }
      |     ^~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:88:5: note:   candidate expects 2 arguments, 1 provided
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:81:14: note: candidate: 'GooString::GooString(std::string&&)'
   81 |     explicit GooString(std::string &&str) : std::string(std::move(str)) { }
      |              ^~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:81:38: note:   no known conversion for argument 1 from 'GooString*' to 'std::string&&' {aka 'std::__cxx11::basic_string<char>&&'}
   81 |     explicit GooString(std::string &&str) : std::string(std::move(str)) { }
      |                        ~~~~~~~~~~~~~~^~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:80:14: note: candidate: 'GooString::GooString(const std::string&)'
   80 |     explicit GooString(const std::string &str) : std::string(str) { }
      |              ^~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:80:43: note:   no known conversion for argument 1 from 'GooString*' to 'const std::string&' {aka 'const std::__cxx11::basic_string<char>&'}
   80 |     explicit GooString(const std::string &str) : std::string(str) { }
      |                        ~~~~~~~~~~~~~~~~~~~^~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:71:14: note: candidate: 'GooString::GooString(const char*)'
   71 |     explicit GooString(const char *sA) : std::string(sA ? sA : "") { }
      |              ^~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:71:36: note:   no known conversion for argument 1 from 'GooString*' to 'const char*'
   71 |     explicit GooString(const char *sA) : std::string(sA ? sA : "") { }
      |                        ~~~~~~~~~~~~^~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:64:5: note: candidate: 'constexpr GooString::GooString(GooString&&)'
   64 |     GooString(GooString &&other) = default;
      |     ^~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:64:27: note:   no known conversion for argument 1 from 'GooString*' to 'GooString&&'
   64 |     GooString(GooString &&other) = default;
      |               ~~~~~~~~~~~~^~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:59:5: note: candidate: 'constexpr GooString::GooString()'
   59 |     GooString() = default;
      |     ^~~~~~~~~
/nix/store/yzmj2l5jhjgryyv74gsm6967i068n19l-poppler-glib-25.10.0-dev/include/poppler/goo/GooString.h:59:5: note:   candidate expects 0 arguments, 1 provided
[CXX] workdir/UnpackedTarball/boost/libs/locale/src/boost/locale/std/std_backend.cpp
[CXX] workdir/UnpackedTarball/boost/libs/locale/src/boost/locale/util/codecvt_converter.cpp
make[1]: *** [/build/libreoffice-25.2.6.2/solenv/gbuild/LinkTarget.mk:336: /build/libreoffice-25.2.6.2/workdir/CxxObject/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:302: build] Error 2
```

### Additional context

_No response_

### System metadata

fetching path input 'path:/nix/store/jd3y5kfy3ywv9lc229nz8vxqbrcyv93d-source'
 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.5, NixOS, 25.11 (Xantusia), 25.11.20251025.6a08e6b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: ~/.config/nix/nix.conf:/nix/store/zgqkd4v11f3h46nwj6jbqyh1zp0izk1y-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:~/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:~/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/<user>/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/sja6jggc51h6cgq6iqffrjykh2lx6hw1-lix-2.93.3/share`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/store/jd3y5kfy3ywv9lc229nz8vxqbrcyv93d-source`

### Notify maintainers


@7c6f434c
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/office/libreoffice/default.nix
+++ b/pkgs/applications/office/libreoffice/default.nix
@@ -313,7 +313,8 @@ stdenv.mkDerivation (finalAttrs: {
 
     # Revert part of https://github.com/LibreOffice/core/commit/6f60670877208612b5ea320b3677480ef6508abb that broke zlib linking
     ./readd-explicit-zlib-link.patch
-
+  ]
+  ++ lib.optionals (lib.versionOlder version "25.8.2.1") [
     # Backport patch to fix build with Poppler 25.09
     (fetchpatch2 {
       url = "https://github.com/LibreOffice/core/commit/7848e02819c007026952a3fdc9da0961333dc079.patch";
@@ -321,10 +322,10 @@ stdenv.mkDerivation (finalAttrs: {
       hash = "sha256-Nw6GFmkFy13w/ktCxw5s7SHL34auP1BQ9JvQnQ65aVU=";
     })
   ]
-  ++ lib.optionals (lib.versionAtLeast version "25.8") [
+  ++ lib.optionals (lib.versionAtLeast version "25.8.2.2") [
     # Fix build with Poppler 25.10
     (fetchpatch2 {
-      url = "https://gitlab.archlinux.org/archlinux/packaging/packages/libreoffice-fresh/-/raw/41e58e117c356af2be83993595caf61f9f30cc89/poppler-25.10.patch";
+      url = "https://gitlab.archlinux.org/archlinux/packaging/packages/libreoffice-fresh/-/raw/f7b0e4385108b95c134599502a7bccf0a41925c8/poppler-25.10.patch";
       hash = "sha256-KMsjDtRRH8Vy/FXaVwxUo0Ww10PCE0sK8+ZL0Ja2kJQ=";
     })
   ]

--- a/pkgs/applications/office/libreoffice/default.nix
+++ b/pkgs/applications/office/libreoffice/default.nix
@@ -321,6 +321,11 @@ stdenv.mkDerivation (finalAttrs: {
       includes = [ "sdext/*" ];
       hash = "sha256-Nw6GFmkFy13w/ktCxw5s7SHL34auP1BQ9JvQnQ65aVU=";
     })
+    # Fix build with Poppler 25.10
+    (fetchpatch2 {
+      url = "https://gitlab.archlinux.org/archlinux/packaging/packages/libreoffice-still/-/raw/f5241554e4a0f6fd95ac4e5cc398a30243407e6a/fix_build_with_poppler_25.10.patch";
+      hash = "sha256-lbPOkc1HeT5Qsp6XfVyVJtmvSL68qTrmbd3q9lvKSu8=";
+    })
   ]
   ++ lib.optionals (lib.versionAtLeast version "25.8.2.2") [
     # Fix build with Poppler 25.10

--- a/pkgs/kde/gear/calligra/default.nix
+++ b/pkgs/kde/gear/calligra/default.nix
@@ -1,6 +1,7 @@
 {
   mkKdeDerivation,
   lib,
+  fetchpatch,
   boost,
   eigen,
   gsl,
@@ -25,6 +26,14 @@
 
 mkKdeDerivation {
   pname = "calligra";
+
+  patches = [
+    # Fix build with Poppler 25.10
+    (fetchpatch {
+      url = "https://invent.kde.org/office/calligra/-/commit/45e8b302bce1d318f310ea13599d7ce84acc477e.patch";
+      hash = "sha256-TECB3eo24+gI8TXL8gw9BIdFWqw0JBKCWpoNVqBSan8=";
+    })
+  ];
 
   extraBuildInputs = [
     boost

